examples/common/synchronous_client.py

```diff
@@ -99,15 +109,28 @@ def run_sync_client():
     # Keep both of these cases in mind when testing as the following will
     # _only_ pass with the supplied asynchronous modbus server (script supplied).
     # ----------------------------------------------------------------------- #
-    log.debug("Write to a Coil and read back")
+    
+    print()
```

Comment:
Why the empty print ? this goes to stdout, whereas log.debug() is passed through the log setup.

If you really want new lines then add it to the log.debug message.

Reply:
It makes it easier to read, by breaks the output in to smaller more to digestible parts. Bellow is output with and without spacing.

### With spacing 
````
2022-07-21 08:13:05,552 MainThread      DEBUG    synchronous_client:79       ### Connecting to server

2022-07-21 08:13:05,552 MainThread      DEBUG    synchronous_client:91       ### Reading Coils
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :127      Current transaction state - IDLE
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :131      Running transaction 1
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x1
2022-07-21 08:13:05,552 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x1 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,553 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x1 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,553 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :495      Adding transaction 1
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :507      Getting transaction 1
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:95       ### printing the content of Reference Number:(index) 1 and  1 bit depth  
2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:96       True

2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:114      ### Test 1 Coil (output): Write 'TRUE' to address:0
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :131      Running transaction 2
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,554 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,555 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,555 MainThread      DEBUG    factory        :347      Factory Response[WriteSingleCoilResponse': 5]
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :495      Adding transaction 2
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :507      Getting transaction 2
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,555 MainThread      DEBUG    synchronous_client:118      ### Test 1 Coil (output): Reading back address:0
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :131      Running transaction 3
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x3 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x0 0x0 0x1
2022-07-21 08:13:05,555 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x3 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,556 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x3 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,556 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :495      Adding transaction 3
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :507      Getting transaction 3
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,556 MainThread      DEBUG    synchronous_client:121      ### address:0 is:True

2022-07-21 08:13:05,556 MainThread      DEBUG    synchronous_client:128      ### Test 2 Coil (output) multiple: Write True to 1-8  coils
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :131      Running transaction 4
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x4 0x0 0x0 0x0 0x8 0x1 0xf 0x0 0x1 0x0 0x8 0x1 0xff
2022-07-21 08:13:05,556 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x4 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,557 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x4 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,557 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleCoilsResponse': 15]
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :495      Adding transaction 4
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :507      Getting transaction 4
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,557 MainThread      DEBUG    synchronous_client:133      ### Test 2 Coil (output) multiple: Reading back address:1-21
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :131      Running transaction 5
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x5 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x15
2022-07-21 08:13:05,558 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x5 0x0 0x0 0x0 0x6 0x1 0x1 0x3 0xff 0xff 0x1f
2022-07-21 08:13:05,558 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x5 0x0 0x0 0x0 0x6 0x1 0x1 0x3 0xff 0xff 0x1f
2022-07-21 08:13:05,558 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :495      Adding transaction 5
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :507      Getting transaction 5
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,558 MainThread      DEBUG    synchronous_client:147      ### Test 3 Coil (output) multiple: Write False to address 1-8  coils
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :131      Running transaction 6
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x6 0x0 0x0 0x0 0x8 0x1 0xf 0x0 0x1 0x0 0x8 0x1 0x0
2022-07-21 08:13:05,559 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x6 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x6 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleCoilsResponse': 15]
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :495      Adding transaction 6
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :507      Getting transaction 6
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,559 MainThread      DEBUG    synchronous_client:152      ### Test 3 Coil (output) multiple: Reading back multiple address:1-8
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :131      Running transaction 7
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x7 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x7 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x0
2022-07-21 08:13:05,560 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x7 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x0
2022-07-21 08:13:05,560 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :495      Adding transaction 7
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :507      Getting transaction 7
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,560 MainThread      DEBUG    synchronous_client:159      ### Test 4 discrete inputs: Read address:0-7 
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :131      Running transaction 8
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x8 0x0 0x0 0x0 0x6 0x1 0x2 0x0 0x0 0x0 0x8
2022-07-21 08:13:05,560 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x8 0x0 0x0 0x0 0x4 0x1 0x2 0x1 0xff
2022-07-21 08:13:05,561 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x8 0x0 0x0 0x0 0x4 0x1 0x2 0x1 0xff
2022-07-21 08:13:05,561 MainThread      DEBUG    factory        :347      Factory Response[ReadDiscreteInputsResponse': 2]
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :495      Adding transaction 8
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :507      Getting transaction 8
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,561 MainThread      DEBUG    synchronous_client:162      ### address 0-7 is:[True, True, True, True, True, True, True, True]

2022-07-21 08:13:05,561 MainThread      DEBUG    synchronous_client:166      ### Test 5 register (Output): Write '10 'to address 1 of registers 
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :131      Running transaction 9
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,561 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,562 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,562 MainThread      DEBUG    factory        :347      Factory Response[WriteSingleRegisterResponse': 6]
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :495      Adding transaction 9
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :507      Getting transaction 9
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,562 MainThread      DEBUG    synchronous_client:170      ### Test 5 register (Output): reading address 1 of registers
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :131      Running transaction 10
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xa 0x0 0x0 0x0 0x6 0x1 0x3 0x0 0x1 0x0 0x1
2022-07-21 08:13:05,562 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 11 bytes !!!!
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xa 0x0 0x0 0x0 0x5 0x1 0x3 0x2 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xa 0x0 0x0 0x0 0x5 0x1 0x3 0x2 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    factory        :347      Factory Response[ReadHoldingRegistersResponse': 3]
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :495      Adding transaction 10
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :507      Getting transaction 10
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:175      ### address 1 is:10

2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:180      ### Test 6 registers multiple: Write '10' to 8 registers
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :131      Running transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xb 0x0 0x0 0x0 0x17 0x1 0x10 0x0 0x1 0x0 0x8 0x10 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xb 0x0 0x0 0x0 0x6 0x1 0x10 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,563 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xb 0x0 0x0 0x0 0x6 0x1 0x10 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,563 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleRegistersResponse': 16]
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :495      Adding transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :507      Getting transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:183      ### Test 6 registers multiple: reading address 1-8 registers
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :131      Running transaction 12
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xc 0x0 0x0 0x0 0x6 0x1 0x3 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,563 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xc 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa
2022-07-21 08:13:05,564 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xc 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa
2022-07-21 08:13:05,564 MainThread      DEBUG    factory        :347      Factory Response[ReadHoldingRegistersResponse': 3]
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :495      Adding transaction 12
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :507      Getting transaction 12
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,564 MainThread      DEBUG    synchronous_client:187      ### address 1-8 is:[10, 10, 10, 10, 10, 10, 10, 10]

2022-07-21 08:13:05,564 MainThread      DEBUG    synchronous_client:191      ### Test 7 Read input registers
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :131      Running transaction 13
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xd 0x0 0x0 0x0 0x6 0x1 0x4 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,564 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xd 0x0 0x0 0x0 0x13 0x1 0x4 0x10 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11
2022-07-21 08:13:05,565 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xd 0x0 0x0 0x0 0x13 0x1 0x4 0x10 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11
2022-07-21 08:13:05,565 MainThread      DEBUG    factory        :347      Factory Response[ReadInputRegistersResponse': 4]
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :495      Adding transaction 13
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :507      Getting transaction 13
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"

2022-07-21 08:13:05,565 MainThread      DEBUG    synchronous_client:204      ### Test 8 Read write registers simultaneously: write
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :131      Running transaction 14
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xe 0x0 0x0 0x0 0x1b 0x1 0x17 0x0 0x1 0x0 0x8 0x0 0x1 0x0 0x8 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,565 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xe 0x0 0x0 0x0 0x13 0x1 0x17 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,583 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xe 0x0 0x0 0x0 0x13 0x1 0x17 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,583 MainThread      DEBUG    factory        :347      Factory Response[ReadWriteMultipleRegistersResponse': 23]
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :495      Adding transaction 14
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :507      Getting transaction 14
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:207      ### Test 8 readwrite result: address 1-8 is:[256, 128, 100, 50, 25, 10, 5, 1]
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :131      Running transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xf 0x0 0x0 0x0 0x6 0x1 0x3 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,584 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xf 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,584 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xf 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,584 MainThread      DEBUG    factory        :347      Factory Response[ReadHoldingRegistersResponse': 3]
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :495      Adding transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :507      Getting transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:209      ### Test 8 read result: address 1-8 is:[256, 128, 100, 50, 25, 10, 5, 1]

2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:220      ### End of Program
````


### Without spacing 
````
2022-07-21 08:13:05,552 MainThread      DEBUG    synchronous_client:79       ### Connecting to server
2022-07-21 08:13:05,552 MainThread      DEBUG    synchronous_client:91       ### Reading Coils
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :127      Current transaction state - IDLE
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :131      Running transaction 1
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x1 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x1
2022-07-21 08:13:05,552 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,552 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x1 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,553 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x1 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x1
2022-07-21 08:13:05,553 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :495      Adding transaction 1
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :507      Getting transaction 1
2022-07-21 08:13:05,553 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:95       ### printing the content of Reference Number:(index) 1 and  1 bit depth  
2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:96       True
2022-07-21 08:13:05,553 MainThread      DEBUG    synchronous_client:114      ### Test 1 Coil (output): Write 'TRUE' to address:0
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :131      Running transaction 2
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,554 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,554 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,555 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x2 0x0 0x0 0x0 0x6 0x1 0x5 0x0 0x0 0xff 0x0
2022-07-21 08:13:05,555 MainThread      DEBUG    factory        :347      Factory Response[WriteSingleCoilResponse': 5]
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :495      Adding transaction 2
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :507      Getting transaction 2
2022-07-21 08:13:05,555 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :495      Adding transaction 3
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :507      Getting transaction 3
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,556 MainThread      DEBUG    synchronous_client:121      ### address:0 is:True
2022-07-21 08:13:05,556 MainThread      DEBUG    synchronous_client:128      ### Test 2 Coil (output) multiple: Write True to 1-8  coils
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :131      Running transaction 4
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x4 0x0 0x0 0x0 0x8 0x1 0xf 0x0 0x1 0x0 0x8 0x1 0xff
2022-07-21 08:13:05,556 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,556 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x4 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,557 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x4 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,557 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleCoilsResponse': 15]
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :495      Adding transaction 4
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :507      Getting transaction 4
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,557 MainThread      DEBUG    synchronous_client:133      ### Test 2 Coil (output) multiple: Reading back address:1-21
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,557 MainThread      DEBUG    transaction    :131      Running transaction 5
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x5 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x15
2022-07-21 08:13:05,558 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,558 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x6 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x6 0x0 0x0 0x0 0x6 0x1 0xf 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleCoilsResponse': 15]
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :495      Adding transaction 6
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :507      Getting transaction 6
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,559 MainThread      DEBUG    synchronous_client:152      ### Test 3 Coil (output) multiple: Reading back multiple address:1-8
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :131      Running transaction 7
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x7 0x0 0x0 0x0 0x6 0x1 0x1 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,559 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,559 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x7 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x0
2022-07-21 08:13:05,560 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x7 0x0 0x0 0x0 0x4 0x1 0x1 0x1 0x0
2022-07-21 08:13:05,560 MainThread      DEBUG    factory        :347      Factory Response[ReadCoilsResponse': 1]
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :495      Adding transaction 7
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :507      Getting transaction 7
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,560 MainThread      DEBUG    synchronous_client:159      ### Test 4 discrete inputs: Read address:0-7 
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :131      Running transaction 8
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x8 0x0 0x0 0x0 0x6 0x1 0x2 0x0 0x0 0x0 0x8
2022-07-21 08:13:05,560 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,560 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 10 bytes !!!!
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x8 0x0 0x0 0x0 0x4 0x1 0x2 0x1 0xff
2022-07-21 08:13:05,561 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x8 0x0 0x0 0x0 0x4 0x1 0x2 0x1 0xff
2022-07-21 08:13:05,561 MainThread      DEBUG    factory        :347      Factory Response[ReadDiscreteInputsResponse': 2]
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :495      Adding transaction 8
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :507      Getting transaction 8
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,561 MainThread      DEBUG    synchronous_client:162      ### address 0-7 is:[True, True, True, True, True, True, True, True]
2022-07-21 08:13:05,561 MainThread      DEBUG    synchronous_client:166      ### Test 5 register (Output): Write '10 'to address 1 of registers 
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :131      Running transaction 9
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :283      SEND: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,561 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,561 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :319      RECV: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,562 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0x9 0x0 0x0 0x0 0x6 0x1 0x6 0x0 0x1 0x0 0xa
2022-07-21 08:13:05,562 MainThread      DEBUG    factory        :347      Factory Response[WriteSingleRegisterResponse': 6]
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :495      Adding transaction 9
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :507      Getting transaction 9
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,562 MainThread      DEBUG    synchronous_client:170      ### Test 5 register (Output): reading address 1 of registers
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :131      Running transaction 10
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xa 0x0 0x0 0x0 0x6 0x1 0x3 0x0 0x1 0x0 0x1
2022-07-21 08:13:05,562 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 11 bytes !!!!
2022-07-21 08:13:05,562 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xa 0x0 0x0 0x0 0x5 0x1 0x3 0x2 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xa 0x0 0x0 0x0 0x5 0x1 0x3 0x2 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    factory        :347      Factory Response[ReadHoldingRegistersResponse': 3]
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :495      Adding transaction 10
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :507      Getting transaction 10
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:175      ### address 1 is:10
2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:180      ### Test 6 registers multiple: Write '10' to 8 registers
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :131      Running transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xb 0x0 0x0 0x0 0x17 0x1 0x10 0x0 0x1 0x0 0x8 0x10 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa 0x0 0xa
2022-07-21 08:13:05,563 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 12 bytes !!!!
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xb 0x0 0x0 0x0 0x6 0x1 0x10 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,563 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xb 0x0 0x0 0x0 0x6 0x1 0x10 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,563 MainThread      DEBUG    factory        :347      Factory Response[WriteMultipleRegistersResponse': 16]
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :495      Adding transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :507      Getting transaction 11
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,563 MainThread      DEBUG    synchronous_client:183      ### Test 6 registers multiple: reading address 1-8 registers
2022-07-21 08:13:05,563 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,564 MainThread      DEBUG    synchronous_client:187      ### address 1-8 is:[10, 10, 10, 10, 10, 10, 10, 10]
2022-07-21 08:13:05,564 MainThread      DEBUG    synchronous_client:191      ### Test 7 Read input registers
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :131      Running transaction 13
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xd 0x0 0x0 0x0 0x6 0x1 0x4 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,564 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,564 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xd 0x0 0x0 0x0 0x13 0x1 0x4 0x10 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11
2022-07-21 08:13:05,565 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xd 0x0 0x0 0x0 0x13 0x1 0x4 0x10 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11 0x0 0x11
2022-07-21 08:13:05,565 MainThread      DEBUG    factory        :347      Factory Response[ReadInputRegistersResponse': 4]
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :495      Adding transaction 13
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :507      Getting transaction 13
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,565 MainThread      DEBUG    synchronous_client:204      ### Test 8 Read write registers simultaneously: write
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :131      Running transaction 14
2022-07-21 08:13:05,565 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xe 0x0 0x0 0x0 0x1b 0x1 0x17 0x0 0x1 0x0 0x8 0x0 0x1 0x0 0x8 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,565 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xe 0x0 0x0 0x0 0x13 0x1 0x17 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,583 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xe 0x0 0x0 0x0 0x13 0x1 0x17 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,583 MainThread      DEBUG    factory        :347      Factory Response[ReadWriteMultipleRegistersResponse': 23]
2022-07-21 08:13:05,583 MainThread      DEBUG    transaction    :495      Adding transaction 14
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :507      Getting transaction 14
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:207      ### Test 8 readwrite result: address 1-8 is:[256, 128, 100, 50, 25, 10, 5, 1]
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :127      Current transaction state - TRANSACTION_COMPLETE
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :131      Running transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :283      SEND: 0x0 0xf 0x0 0x0 0x0 0x6 0x1 0x3 0x0 0x1 0x0 0x8
2022-07-21 08:13:05,584 MainThread      DEBUG    sync           :77       New Transaction state "SENDING"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :304      Changing transaction state from "SENDING" to "WAITING FOR REPLY"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :402      {msg_start} received, Expected 1024 bytes Received 25 bytes !!!!
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :408      Changing transaction state from "WAITING FOR REPLY" to "PROCESSING REPLY"
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :319      RECV: 0x0 0xf 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,584 MainThread      DEBUG    socket_framer  :165      Processing: 0x0 0xf 0x0 0x0 0x0 0x13 0x1 0x3 0x10 0x1 0x0 0x0 0x80 0x0 0x64 0x0 0x32 0x0 0x19 0x0 0xa 0x0 0x5 0x0 0x1
2022-07-21 08:13:05,584 MainThread      DEBUG    factory        :347      Factory Response[ReadHoldingRegistersResponse': 3]
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :495      Adding transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :507      Getting transaction 15
2022-07-21 08:13:05,584 MainThread      DEBUG    transaction    :229      Changing transaction state from "PROCESSING REPLY" to "TRANSACTION_COMPLETE"
2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:209      ### Test 8 read result: address 1-8 is:[256, 128, 100, 50, 25, 10, 5, 1]
2022-07-21 08:13:05,584 MainThread      DEBUG    synchronous_client:220      ### End of Program
````

Reply:
it is a matter of taste what is better.  BUT mixing print() and log.debug() is a bad idea. Generally log output are one liners making it easier to grasp in the log, but if you really must, then add the new line to log.debug

Reply:
It sound like the examples are used for testing and using.

If is used for testing, maybe make a copy to test? Were one can remove any "noob". friendliness?

Reply:
I think it is the other way round, many tests are copied to examples without being modified.

We do not use the examples directory for testing.